cfme/configure/configuration/analysis_profile.py

```diff
@@ -0,0 +1,440 @@
+        self.description = description
+        self.files = files if isinstance(files, (list, type(None))) else [files]
+        self.events = events if isinstance(events, (list, type(None))) else [events]
+        self.categories = categories if isinstance(categories, (list, type(None))) else [categories]
+        self.registry = registry if isinstance(registry, (list, type(None))) else [registry]
+        if profile_type in (self.VM_TYPE, self.HOST_TYPE):
+            self.profile_type = profile_type
+        else:
+            raise ValueError("Profile Type is incorrect")
+
+    def create(self, cancel=False):
+        """Add Analysis Profile to appliance"""
+        # The tab form values have to be dictionaries with the root key matching the tab widget name
+        form_values = self.form_fill_args()
+
+        view = navigate_to(self, 'Add')
+        view.form.fill(form_values)
+
+        if cancel:
+            view.cancel.click()
+        else:
+            view.add.click()
+
+        view = self.create_view(AnalysisProfileAllView)
+        view.flush_widget_cache()
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        view.flash.assert_success_message(
+            'Add of new Analysis Profile was cancelled by the user'
+            if cancel
+            else 'Analysis Profile "{}" was saved'.format(self.name))
+
+    def update(self, updates, cancel=False):
+        """Update the existing Analysis Profile with given updates dict
+        Make use of Updateable and use `with` to update object as well
+        Note the updates dict should take the structure below if called directly
+
+            .. code-block:: python
+                updates = {
+                    'name': self.name,
+                    'description': self.description,
+                    'files': {
+                        'tab_form': ['/example/file']},
+                    'events': {
+                        'tab_form': ['example_event']},
+                    'categories': {
+                        'tab_form': ['Example']},
+                    'registry': {
+                        'tab_form': ['example_registry']}
+                }
+
+            Args:
+                updates (dict): Dictionary of values to change in the object.
+                cancel (boolean): whether to cancel the update
+        """
+        # hack to work around how updates are passed when used in context mgr
+        # TODO revisit this method when BZ is fixed:
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1485953
+        form_fill_args = self.form_fill_args(updates=updates)
+        view = navigate_to(self, 'Edit')
+        changed = view.form.fill(form_fill_args)
+
+        if changed and not cancel:  # save button won't be enabled if nothing was changed
+            view.save.click()
+        else:
+            view.cancel.click()
+
+        view = self.create_view(AnalysisProfileDetailsView)  # edited through details
```

Comment:
**REQUIRED**
```python
view = self.create_view(AnalysisProfileDetailsView, override=updates)
```

Reply:
Testing

Reply:
Looks like it's working now - when I had tried before I think I had other bugs preventing it from working correctly.

Awesome! Thanks @quarckster